sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/options/PipelineOptionsFactory.java

```diff
@@ -673,32 +677,18 @@ public static void printHelp(PrintStream out, Class<? extends PipelineOptions> i
     Preconditions.checkNotNull(iface);
     validateWellFormed(iface, REGISTERED_OPTIONS);
 
-    Iterable<Method> methods =
-        Iterables.filter(
-            ReflectHelpers.getClosureOfMethodsOnInterface(iface), NOT_SYNTHETIC_PREDICATE);
-    ListMultimap<Class<?>, Method> ifaceToMethods = ArrayListMultimap.create();
-    for (Method method : methods) {
-      // Process only methods that are not marked as hidden.
-      if (method.getAnnotation(Hidden.class) == null) {
-        ifaceToMethods.put(method.getDeclaringClass(), method);
-      }
+    Set<PipelineOptionsReflector.Property> properties =
+        PipelineOptionsReflector.collectVisibleProperties(iface);
+
+    RowSortedTable<Class<?>, String, Method> ifacePropGetterTable = TreeBasedTable.create(
+        ClassNameComparator.INSTANCE, Ordering.natural());
+    for (PipelineOptionsReflector.Property prop : properties) {
+      ifacePropGetterTable.put(prop.definingInterface(), prop.name(), prop.getterMethod());
     }
-    SortedSet<Class<?>> ifaces = new TreeSet<>(ClassNameComparator.INSTANCE);
-    // Keep interfaces that are not marked as hidden.
-    ifaces.addAll(Collections2.filter(ifaceToMethods.keySet(), new Predicate<Class<?>>() {
-      @Override
-      public boolean apply(Class<?> input) {
-        return input.getAnnotation(Hidden.class) == null;
-      }
-    }));
-    for (Class<?> currentIface : ifaces) {
-      Map<String, Method> propertyNamesToGetters =
-          getPropertyNamesToGetters(ifaceToMethods.get(currentIface));
 
-      // Don't output anything if there are no defined options
-      if (propertyNamesToGetters.isEmpty()) {
-        continue;
-      }
+    for (Class<?> currentIface : ifacePropGetterTable.rowKeySet()) {
```

Comment:
rowKeySet -> rowMap().entrySet()

Then you can get the iface and map by calling entry.getKey()/entry.getValue()

Iterating over the entryset is typically much faster then keyset + lookup